pkg/sql/sem/builtins/pg_builtins.go

```diff
@@ -1503,7 +1498,15 @@ SELECT description
 				return nil, err
 			}
 			return runPrivilegeChecks(privs, func(priv privilege.Privilege) (tree.Datum, error) {
-				return hasPrivilege(ctx, specifier, user, priv)
+				ret, err := ctx.Planner.HasPrivilege(ctx.Context, specifier, user, priv)
+				if err != nil {
+					// When a TableOID is specified and the relation is not found, we return NULL.
+					if specifier.TableOID != nil && sqlerrors.IsUndefinedRelationError(err) {
```

Comment:
this is duplicated in a few places. can we introduce a helper method:

```
function handleHasPrivilegeError(specifier ..., err error) (tree.Datum, error) {
   if (specifier.TableOID != nil || specifier.DatabaseOID != nil) && sqlerrors.IsUndefinedRelationError(err) {
     return tree.DNull, nil
   }
   return nil, err
}
```

and these be replaced with
```
return handleHasPrivilegeError(specifier, error)
```